### ISSUE TITLE
How to train a network with multiple output layers

### ISSUE THREAD

**Issue by Arkoprabho** [Microsoft/CNTK#2998]
Objective: To train a neural network as shown in the image below, for an image classification task. (Input dimension can be seen in the image.)
![](https://i.stack.imgur.com/KdaZ2.png)

I have been able to create the network, but am facing problems below:
1. Specifying the loss and classification error functions.
2. Specifying the output variable to be fed to `train_minibatch` dictionary.

I am using a generator to feed the data to the network, and thus using `train_minibatch` to feed the data. The output of this generator are the input images, and a list of 4 items containing the 1 hot encoded labels.

**Comment by KeDengMS** [comment]
1. A straight-forward implementation of loss/classification error would be compute loss and classification error for each output, then sum up for the 4 outputs. This design highly depends on what you want to optimize for.
2. You can pass the model to train_minibatch using [combine](https://cntk.ai/pythondocs/cntk.ops.html#cntk.ops.combine).

**Comment by Arkoprabho** [comment]
```
loss0 = cntk.cross_entropy_with_softmax(z.outputs[0], output_placeholder)
loss1 = cntk.cross_entropy_with_softmax(z.outputs[1], output_placeholder)
loss2 = cntk.cross_entropy_with_softmax(z.outputs[2], output_placeholder)
loss3 = cntk.cross_entropy_with_softmax(z.outputs[3], output_placeholder)

loss = loss0 + loss1 + loss2 + loss3

label_error1 = cntk.classification_error(z.outputs[0], output_placeholder)
label_error2 = cntk.classification_error(z.outputs[1], output_placeholder)
label_error3 = cntk.classification_error(z.outputs[2], output_placeholder)
label_error4 = cntk.classification_error(z.outputs[3], output_placeholder)

label_error = label_error1 + label_error2 + label_error3 + label_error4
```

Something like this?
How should i structure the output placeholder? And then feed it to the argument dict of train_minibatch?

**Comment by Arkoprabho** [comment]
I wish to replicate the ideas and methodology as seen in [this link](https://github.com/ypwhs/captcha_break/blob/master/cnn.ipynb) using CNTK rather than Keras

**Comment by Arkoprabho** [comment]
So i had to add 4 different output variables and feed data to them separately. Thank you @KeDengMS I will be updating the code at my [repo](https://github.com/Arkoprabho/BSCaptcha)
